crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,651 @@
+        // use the standard message for other clients?
+        consumer_chains
+            .iter()
+            .any(|(chain_id, _)| chain_id == chain.id())
+    } else {
+        false
+    }
+}
+
+/// Fetch all the counterparty clients of the given chain.
+/// A counterparty client is a client that has a connection with that chain.
+///
+/// 1. Fetch all connections on the given chain
+/// 2. For each connection:
+///     2.1. Fetch the client state of the counterparty client of that connection.
+///     2.2. From the client state, extract the chain id of the counterparty chain.
+/// 4. Return a list of all counterparty chains and counterparty clients.
+fn fetch_all_counterparty_clients(
+    chain: &CosmosSdkChain,
+) -> eyre::Result<Vec<(ChainId, ClientId)>> {
+    use ibc_relayer::chain::requests::PageRequest;
+    use ibc_relayer::chain::requests::{QueryClientStateRequest, QueryConnectionsRequest};
+
+    let connections = chain.query_connections(QueryConnectionsRequest {
+        pagination: Some(PageRequest::all()),
+    })?;
+
+    debug!("found {} connections", connections.len());
+
+    let mut counterparty_clients = vec![];
+
+    for connection in connections {
+        debug!(
+            "fetching counterparty client state for connection {}",
+            connection.connection_id
+        );
+
+        let client_id = connection.connection_end.client_id();
+
+        let client_state = chain.query_client_state(
+            QueryClientStateRequest {
+                client_id: client_id.clone(),
+                height: QueryHeight::Latest,
+            },
+            IncludeProof::No,
+        );
+
+        let client_state = match client_state {
+            Ok((client_state, _)) => client_state,
+            Err(e) => {
+                warn!(
+                    "failed to fetch client state for counterparty client {client_id}, skipping..."
+                );
+                warn!("reason: {e}");
+
+                continue;
+            }
+        };
+
+        let counterparty_chain_id = client_state.chain_id();
+        debug!("found counterparty client with id {client_id} on counterparty chain {counterparty_chain_id}");
+
+        counterparty_clients.push((counterparty_chain_id, client_id.clone()));
+    }
+
+    Ok(counterparty_clients)
+}
+
+/// Build the two headers to submit as part of the `MsgSubmitMisbehaviour` message.
+fn build_evidence_headers(
+    rt: Arc<TokioRuntime>,
+    chain: &CosmosSdkChain,
+    lc: LightClientAttackEvidence,
+) -> eyre::Result<(TendermintHeader, TendermintHeader)> {
+    let trusted_validator_set = rt
+        .block_on(chain.rpc_client.validators(lc.common_height, Paging::All))?
+        .validators;
+
+    let header1 = {
+        TendermintHeader {
+            signed_header: lc.conflicting_block.signed_header,
+            validator_set: lc.conflicting_block.validator_set,
+            trusted_height: Height::from_tm(lc.common_height, chain.id()),
```

Comment:
FIXME: @sainoe is running into an error where the node rejects the evidence because the trusted height is equal to the header height, ie. the height of the conflicting block is the same as the common height. Maybe the light client detector is doing something wrong?

cc @ancazamfir

Reply:
I am not sure which script exactly @sainoe was using, but I have myself been using this one and could not reproduce the issue he was seeing: https://github.com/informalsystems/hermes/pull/3456/files#diff-a9944185c7e5bf34c51e2a6f1daec2ef8ef0a5cf4ecb3fb81933cf742cb4bb34